environment-pipelines/codepipeline.tf

```diff
@@ -24,6 +24,20 @@ resource "aws_codepipeline" "environment_pipeline" {
     }
   }
 
+  trigger {
+    provider_type = "CodeStarSourceConnection"
+    git_configuration {
+      source_action_name = "GitCheckout"
+      push {
+        branches {
+          includes = [
+            var.trigger_on_push ? var.branch : "NO_TRIGGER"
```

Comment:
Ideally here we would just not have a trigger instead of setting this placeholder branch to prevent triggering `var.trigger_on_push ? var.branch : "NO_TRIGGER"`, however Terraform doesn't detect the change if we omit the entire trigger block.